src/Application/Asset/TerminalAssetPersistenceService.php

```
@@ -0,0 +1,116 @@
+        }
+
+        throw RepositoryUnavailableException::forReason(self::REPOSITORY_FAILURE_REASON, $exception);
+    }
+
+    private function cacheTerminalStatus(Asset $asset): TerminalStatusCacheStoreResult
+    {
+        try {
+            // Terminal status caching is best-effort after MySQL persistence succeeds.
+            $this->assetTerminalStatusCache->store($asset->getId(), $asset->getStatus());
+        } catch (\Throwable $exception) {
+            return TerminalStatusCacheStoreResult::failed($this->terminalStatusCacheError($exception));
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\App\Application\Asset\Result\TerminalStatusCacheStoreResult' in method 'cacheTerminalStatus'. (undefined)

(StaticAccess)